keps/sig-apps/19-Graduate-CronJob-to-Stable/README.md

```diff
@@ -0,0 +1,503 @@
+
+The targets are defined by the below suggested maximum limits, which are organized the same way as the [Kubernetes native type thresholds](https://github.com/kubernetes/community/blob/b58777eda0f0077eba79d063c1c8f31cd7dbb8b9/sig-scalability/configs-and-limits/thresholds.md#kubernetes-thresholds).
+
+#### CronJob Limits
+
+There should be nothing in the implementation that limits CronJobs per namespace. Overall cluster-wide limits of CronJob are important. Cluster wide limits for CronJob should be storage bound since it shares the storage space with all other objects. Determining the appropriate storage limit for a cluster is out-of-scope for this document.
+
+#### Frequency of launched jobs
+
+The number of CronJobs is also sensitive to the API server QPS and the schedule of the individual CronJobs. This translates to the frequency of launched jobs. We could have large number of CronJobs with a spread of schedule that doesn't stress the Job API. At the same time we could have a small number of CronJobs that schedule synchronously stressing the Jobs API. The design must be able to easily saturate the API server QPS. The user can setup rate limits for CronJob and Job APIs using [API Server rate limting config](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#eventratelimit).
+
+## API changes
+
+These are the new fields added as part of promotion to stable:
+- `.status`
+  - `.lastSuccessfulTime`
+  - `.nextScheduleTime`
+
+### CronJob v1 API
+
+```golang
+// CronJob represents the configuration of a single cron job.
+type CronJob struct {
+	metav1.TypeMeta
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta
+
+	// Specification of the desired behavior of a cron job, including the schedule.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Spec CronJobSpec
+
+	// Current status of a cron job.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Status CronJobStatus
+}
+
+
+// CronJobList is a collection of cron jobs.
+type CronJobList struct {
+	metav1.TypeMeta
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta
+
+	// items is the list of CronJobs.
+	Items []CronJob
+}
+
+// CronJobSpec describes how the job execution will look like and when it will actually run.
+type CronJobSpec struct {
+
+	// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
+	Schedule string
+
+	// Optional deadline in seconds for starting the job if it misses scheduled
```

Comment:
nit: I don't think it matters much for the KEP itself, but golang comments should start with field name.
So if you fix them here it may be simpler to copy-paste the to the code later :)

Reply:
Same in bunch of other places

Reply:
I wouldn't bother that, this is direct copy of the API types.